### PR TITLE
feat: [PL-29958]: added apiUrl prefix in ServiceVersions

### PR DESCRIPTION
**Pull request by mayankVermaHarness**
### Summary

JIRA: https://harness.atlassian.net/browse/PL-29958
Today, all versions are coming from Prod1 cluster. Adding a base URL `/gateway` prefix using `window.getApiBaseUrl` to the individual APIs should fix it—gateway will route the requests to the correct cluster.

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by mayankVermaHarness** [comment]
retrigger jest

**Comment by mayankVermaHarness** [comment]
retrigger titlecheck

File: src/modules/33-auth-settings/pages/AccountOverview/views/ServiceVersions.tsx [hunk 1]
```diff
@@ -118,7 +118,10 @@ const ServiceVersions = () => {
   const fetchServices = () => {
     const servicesLength = data.length
     if (servicesLength <= 0) {
-      const promiseArr = versionAPIs.map(row => fetch(row.url.startsWith('http') ? row.url : BASE_URL + row.url))
+      const promiseArr = versionAPIs.map(row => {
+        const finalUrl = BASE_URL + window.getApiBaseUrl(row.url)
```
Comment (abhinavrastogi-harness) [inline src/modules/33-auth-settings/pages/AccountOverview/views/ServiceVersions.tsx:122]: why is `BASE_URL` needed if we are already using `window.getApiBaseUrl`?
also, why not just use `window.API_URL`?

Comment (mayankVermaHarness) [reply src/modules/33-auth-settings/pages/AccountOverview/views/ServiceVersions.tsx:122]: For 1st Point:
Agreed, we don't need BASE_URL anymore. It is take care of internally by `windowgetApiBaseUrl`. I'm making this change.

For 2nd Point:
`window.apiUrl` can be `undefined` for some environments where—say, gateway isn't deployed.
`window.getApiBaseUrl` is a wrapper that takes care of the above scenario (it simply adds a prefix of `/gateway` if needed)

**Review by abhinavrastogi-harness — APPROVED** [review]

**Review by avinashmadhwani02 — APPROVED** [review]

**Pull request merged**